hello-server.php

```diff
@@ -45,74 +43,184 @@
+ */
+function lw_get_server_ip() {
+
+	// Bail if we don't have a server address.
+	if ( empty( $_SERVER['SERVER_ADDR'] ) ) {
+		return false;
 	}
-	return $serverIp;
+
+	// Fetch our server address.
+	$server_ip  = sanitize_text_field( $_SERVER['SERVER_ADDR'] );
+
+	// If we're on a local, do some additional checks.
+	if ( '127.0.0.1' === $server_ip ) {
+
+		// Get our hostname data.
+		$res_ip   = gethostbyname( gethostname() );
+
+		// Run a validation check on our IP to make sure it's legit.
+		if ( filter_var( $res_ip, FILTER_VALIDATE_IP ) === true ) {
+			return $res_ip;
+		}
+	}
+
+	// Return our IP.
+	return $server_ip;
 }
 
-// This just echoes the chosen line, we'll position it later
-function hello_server( $wp_admin_bar ) {
-	$serverInfo = hello_server_get_info();
-	$mainNode = [
-	  'id' => 'lw_hello_server',
-	  'title' => __("Hello, Server", "hello-server"),
-	  'meta' => ['title' => __('View the current server info.', "hello-server"), 'class' => 'menupop']
-
-	];
-	$hostNode = [
-	  'id' => 'lw_host_server',
-      'title' => "Hello, I'm: <span itemprop='server-name'>".$serverInfo['hostname'] . "</span>!",
-	  'meta' => ['title' => 'View the current server hostname.'],
-	  'parent' => 'lw_hello_server'
-	];
-	$ipNode = [
-	  'id' => 'lw_server_ip',
-	  'title' => 'My IP is: '.$serverInfo['ip'],
-	  'meta' => ['title' => 'The current servers IP Address.'],
-	  'parent' => 'lw_hello_server'
-	];
-	$wp_admin_bar->add_node( $mainNode );
-	$wp_admin_bar->add_node( $hostNode );
-	$wp_admin_bar->add_node( $ipNode );
+/**
+ * Include a small bit of CSS to display and position the icon.
+ *
+ * @return void
+ */
+function lw_add_admin_bar_css() {
+
+	echo '
+	<style>
+
+		li#wp-admin-bar-lw-hello-server {
+			display: block;
+		}
+
+		li#wp-admin-bar-lw-hello-server .ab-item.ab-empty-item .ab-lw-icon {}
+
+		li#wp-admin-bar-lw-hello-server .ab-item.ab-empty-item .ab-lw-icon:before {
+			content: "\f115";
+			top: 2px;
+		}
+
+		li#wp-admin-bar-lw-hello-server .lw-admin-child-menu .ab-item.ab-empty-item span {
+			color: #fff;
+		}
+
+		@media screen and ( max-width: 782px ) {
+
+			li#wp-admin-bar-lw-hello-server {
+				display: block !important;
+			}
+		}
+
+	</style>
+	';
 }
+add_action( 'wp_head', 'lw_add_admin_bar_css' );
+add_action( 'admin_head', 'lw_add_admin_bar_css' );
+
+/**
+ * Display our server info in the admin bar.
+ *
+ * @param  WP_Admin_Bar $wp_admin_bar  The global WP_Admin_Bar object.
+ *
+ * @return void
+ */
+function lw_load_hello_server( WP_Admin_Bar $wp_admin_bar ) {
+
+	// Fetch our server info.
+	$info	= lw_hello_server_get_info();
 
-// Now we set that function up to execute when the admin_notices action is called
-add_action( 'admin_bar_menu', 'hello_server', 999 );
+	// Add a parent item.
+	$wp_admin_bar->add_node(
+		array(
+			'id'    => 'lw-hello-server',
+			'title' => '<span class="ab-icon ab-lw-icon"></span><span class="ab-label ab-lw-label">' . __( 'Hello Server', 'hello-server' ) . '</span>',
```

Comment:
Since this is being used to generate markup, the i18n output should be escaped (e.g. `esc_html__()` instead of `__()` for 'title').